File: src/main/java/no/nav/service/pensjon/regler/to/BeslutningsstotteRequest.java
package no.nav.service.pensjon.regler.to;


import no.nav.domain.pensjon.regler.grunnlag.InntektKontrollGrunnlag;
import no.nav.domain.pensjon.regler.grunnlag.Persongrunnlag;
import no.nav.domain.pensjon.regler.kode.BeslutningsstotteTypeCti;
import no.nav.domain.pensjon.regler.kode.BorMedTypeCti;
import no.nav.domain.pensjon.regler.vedtak.VilkarsVedtak;

/**
 * @author Lars Hartviksen (Decisive), PK-7048
 * @author Swiddy de Louw (Capgemini) - PK-7973 - PKPYTON-526
 * @author Frederik Rønenvig (Decisive) - PK-14979
 */
public class BeslutningsstotteRequest extends ServiceRequest {


    private BeslutningsstotteTypeCti beslutningsstotteType;

    private Persongrunnlag persongrunnlag;

    /** Vilkårsvedtaket for uføretrygd */
    private VilkarsVedtak vilkarsvedtak;

    /** Vurdert sivilstand på Kravhode.konverteringsgrunnlagUT.lopendeUFB.benyttetSivilstand */
    private BorMedTypeCti benyttetSivilstand;

    /** Inneholder nødvendig informasjon for å beregne ny forventet inntekt ifbm løpende inntektskontroll. */
    private InntektKontrollGrunnlag inntektKontrollGrunnlag;


}

File: src/main/kotlin/no/nav/pensjon/regler/to/BeslutningsstotteRequest.kt
package no.nav.pensjon.regler.to

import no.nav.pensjon.regler.domain.enum.BeslutningsstotteTypeEnum
import no.nav.pensjon.regler.domain.enum.BorMedTypeEnum
import no.nav.pensjon.regler.domain.grunnlag.InntektKontrollGrunnlag
import no.nav.pensjon.regler.domain.grunnlag.Persongrunnlag
import no.nav.pensjon.regler.domain.vedtak.VilkarsVedtak

class BeslutningsstotteRequest : ServiceRequest() {
    var beslutningsstotteTypeEnum: BeslutningsstotteTypeEnum? = null
    var persongrunnlag: Persongrunnlag? = null

    /** vilkårsvedtaket for uføretrygd  */
    var vilkarsvedtak: VilkarsVedtak? = null

    /** Vurdert sivilstand på Kravhode.konverteringsgrunnlagUT.lopendeUFB.benyttetSivilstand  */
    var benyttetSivilstandEnum: BorMedTypeEnum? = null

    /** Inneholder nødvendig informasjon for å beregne ny forventet inntekt ifbm løpende inntektskontroll.  */
    var inntektKontrollGrunnlag: InntektKontrollGrunnlag? = null
}